src/uct/ugni/smsg/ugni_smsg_ep.c

```diff
@@ -174,6 +174,15 @@ uct_ugni_smsg_ep_am_common_send(uct_ugni_smsg_ep_t *ep, uct_ugni_smsg_iface_t *i
 {
     gni_return_t gni_rc;
 
+    if(iface->super.send_q_len > 0 && iface->super.send_in_pending == 0) {
```

Comment:
@MattBBaker since it is in the critical path, please "unlike" this, even before we discuss all the rest